examples/animation/ease_functions.rs

```diff
@@ -0,0 +1,178 @@
+                    p.spawn(SpriteBundle {
+                        sprite: Sprite {
+                            custom_size: Some(Vec2::new(5.0, 5.0)),
+                            color,
+                            ..default()
+                        },
+                        transform: Transform::from_xyz(110.0, 15.0, 0.0),
+                        ..default()
+                    });
+                    p.spawn(SpriteBundle {
+                        sprite: Sprite {
+                            custom_size: Some(Vec2::new(4.0, 4.0)),
+                            color,
+                            ..default()
+                        },
+                        transform: Transform::from_xyz(0.0, 0.0, 0.0),
+                        ..default()
+                    });
+                });
+        }
+    }
+    commands.spawn(
+        TextBundle::from_section("", TextStyle::default()).with_style(Style {
+            position_type: PositionType::Absolute,
+            bottom: Val::Px(12.0),
+            left: Val::Px(12.0),
+            ..default()
+        }),
+    );
+}
+
+fn display_curves(
+    mut gizmos: Gizmos,
+    ease_functions: Query<(&SelectedEaseFunction, &Transform, &Children)>,
+    mut transforms: Query<&mut Transform, Without<SelectedEaseFunction>>,
+    mut ui: Query<&mut Text, With<Node>>,
+    time: Res<Time>,
+) {
+    let samples = 100;
+    let size = 100.0;
+    let duration = 2.5;
+    let time_margin = 0.5;
+
+    let now = ((time.elapsed_seconds() % (duration + time_margin * 2.0) - time_margin) / duration)
+        .clamp(0.0, 1.0);
+
+    ui.single_mut().sections[0].value = format!("Progress: {:.2}", now);
+
+    for (SelectedEaseFunction(function, color), transform, children) in &ease_functions {
+        // Draw a box around the curve
+        gizmos.linestrip_2d(
+            [
+                Vec2::new(transform.translation.x, transform.translation.y + 15.0),
+                Vec2::new(
+                    transform.translation.x + size,
+                    transform.translation.y + 15.0,
+                ),
+                Vec2::new(
+                    transform.translation.x + size,
+                    transform.translation.y + 15.0 + size,
+                ),
+                Vec2::new(
+                    transform.translation.x,
+                    transform.translation.y + 15.0 + size,
+                ),
+                Vec2::new(transform.translation.x, transform.translation.y + 15.0),
+            ],
+            color.darker(0.4),
+        );
+
+        // Draw the curve
+        let f = easing::EasingCurve::ease(*function);
+        gizmos.linestrip_2d(
+            (0..(samples + 1)).map(|i| {
+                Vec2::new(
+                    i as f32 / samples as f32 * size + transform.translation.x,
+                    f.sample(i as f32 / samples as f32).unwrap() * size
+                        + transform.translation.y
+                        + 15.0,
+                )
```

Comment:
Maybe the sampled value could be split on another line and in a variable, possibly with a comment?